### PR TITLE
Add Jekyll Cache

### PR DESCRIPTION
**Pull request by typhoon2099**
Thanks for contributing the Netlify plugins directory!

**Are you adding a plugin or updating one?**

- [x] Adding a plugin
- [ ] Updating a plugin

**Have you completed the following?**

- [x] Read and followed the [plugin author guidelines](https://github.com/netlify/plugins/blob/master/docs/guidelines.md).
- [x] Included all [required fields](https://github.com/netlify/plugins/blob/master/docs/CONTRIBUTING.md#required-fields) in your entry.
- [x] Tested the plugin [locally](https://docs.netlify.com/cli/get-started/#run-builds-locally) and [on Netlify](https://docs.netlify.com/configure-builds/build-plugins/#install-a-plugin), using the plugin version stated in your entry.

**Test plan**
Please add a link to a successful public deploy log using the stated version of the plugin. Include any other context reviewers might need for testing.
First build: https://app.netlify.com/sites/matthewsimpson/deploys/60143c9e4b7b62048cbadf4a
Second build: https://app.netlify.com/sites/matthewsimpson/deploys/60143eb5a598010008f44965

The cache is stored in the first and restored in the second, which should improve Jekyll build times. It's probably less noticable on this site but I have other (private) sites that make use of image resizing that benefit greatly from this plugin (a 4m 30s build was reduced to 1m 15s).

### CODE REVIEW

**Review by ehmicky — COMMENTED** [review]
Hi @typhoon2099, thanks a lot for this plugin!

Everything is looking great!

One thing I am noticing in the before/after build logs: although it does appear that the directories are being properly cached then restored, the duration of `jekyll build` does not seem to change. Do you have example of build logs where this duration improves? Thanks!

**Comment by typhoon2099** [comment]
Hi @ehmickey, I can try and do another Couple of builds on something else, my own site doesn't benefit too much from it but another private site does. I need to migrate to a better image processing plugin, so I'll link to those logs when I've finished.

**Comment by ehmicky** [comment]
Thanks @typhoon2099!

**Comment by typhoon2099** [comment]
Cold: https://app.netlify.com/sites/matthewsimpson/deploys/601581cab0149589a9eb56fa
Warm: https://app.netlify.com/sites/matthewsimpson/deploys/601584097e8812abb6967e93
36.9s -> 27.7s
There seems to be a slower build between these, but I'm attributing that down to a server blip. It's not a totally obvious increase (I'm using LSI in Jekyll which is _always_ slow, and don't have a lot of images), but it's certainly reduced.

**Review by ehmicky — APPROVED** [review]

**Comment by ehmicky** [comment]
Thanks @typhoon2099, this is looking good!
After fixing the merge conflict, this can be merged :tada:

**Comment by typhoon2099** [comment]
Should be sorted now.

File: site/plugins.json [hunk 1]
```diff
@@ -437,6 +437,14 @@
     "name": "Gmail",
     "package": "netlify-plugin-gmail",
     "repo": "https://github.com/racoonx2p/netlify-plugin-gmail",
-    "version": "1.1.0"
+    "version": "1.1.0",
```
Comment (ehmicky) [inline site/plugins.json:440]:
```suggestion
    "version": "1.1.0"
```

JSON does not allow trailing commas.

**Comment by typhoon2099** [comment]
Sorry! Messed it up when I moved it about.

**Pull request merged**